content/docs/07-reference/04-stream-helpers/14-google-generative-ai-stream.mdx

```diff
@@ -1,85 +0,0 @@
----
-title: GoogleGenerativeAIStream
-description: Learn to use GoogleGenerativeAIStream helper function in your application.
----
-
-# `GoogleGenerativeAIStream`
-
-<Note type="warning">
-  GoogleGenerativeAIStream has been removed in AI SDK 4.0.
-</Note>
-
-<Note type="warning">
-  GoogleGenerativeAIStream is part of the legacy Google Generative AI
-  integration. It is not compatible with the AI SDK 3.1 functions. It is
-  recommended to use the [AI SDK Google Generative AI
-  Provider](/providers/ai-sdk-providers/google-generative-ai) instead.
-</Note>
-
-The GoogleGenerativeAIStream function is a utility that transforms the output from Google's Generative AI SDK into a ReadableStream. It uses AIStream under the hood, applying a specific parser for the Google's response data structure. This works with the official Generative AI SDK, and it's supported in both Node.js, Edge Runtime, and browser environments.
-
-## Import
-
-### React
-
-<Snippet
-  text={`import { GoogleGenerativeAIStream } from "ai"`}
-  prompt={false}
-/>
-
-## API Signature
-
-### Parameters
-
-<PropertiesTable
-  content={[
-    {
-      name: 'response',
-      type: '{ stream: AsyncIterable<GenerateContentResponse> }',
-      description:
-        'The response object returned by the Google Generative AI API.',
-    },
-    {
-      name: 'callbacks',
-      type: 'AIStreamCallbacksAndOptions',
-      isOptional: true,
-      description:
-        'An object containing callback functions to handle the start, each token, and completion of the AI response. In the absence of this parameter, default behavior is implemented.',
-      properties: [
-        {
-          type: 'AIStreamCallbacksAndOptions',
-          parameters: [
-            {
-              name: 'onStart',
-              type: '() => Promise<void>',
-              description:
-                'An optional function that is called at the start of the stream processing.',
-            },
-            {
-              name: 'onCompletion',
-              type: '(completion: string) => Promise<void>',
-              description:
-                "An optional function that is called for every completion. It's passed the completion as a string.",
-            },
-            {
-              name: 'onFinal',
-              type: '(completion: string) => Promise<void>',
-              description:
-                'An optional function that is called once when the stream is closed with the final completion message.',
-            },
-            {
-              name: 'onToken',
-              type: '(token: string) => Promise<void>',
-              description:
-                "An optional function that is called for each token in the stream. It's passed the token as a string.",
-            },
-          ],
-        },
-      ],
-    },
-  ]}
-/>
-
-### Returns
-
-A `ReadableStream`.
```

Comment:
pls leave - this needs to be removed separately

Reply:
I was searching through the codebase and i stumbled across it was already removed:


![CleanShot 2025-04-24 at 13 56 48@2x](https://github.com/user-attachments/assets/28d63214-dc99-48f3-a210-a962113bea7f)

so i did what you suggested in the last one which was to remove it

Reply:
can re-add back if you want, maybe we need to re-add the other one too if we do?

Reply:
no just the docs - the whole section will eventually be removed